docs/styleguide/PyGuide.md

```diff
@@ -236,6 +237,38 @@ with open("hello.txt") as hello_file:
         print(line)
 ```
 
+
+<a id="s2.9-abstract-classes"></a>
+<a id="29-abstract-classes"></a>
+<a id="abstract-classes"></a>
+### 2.9 Abstract Classes 
+
+When defining abstract classes, the following template should be used:
+
+```python
+from abc import ABC, abstractmethod
+
+class AccuracyMetric(ABC):
+    """
+    Base abstract class for accuracy metrics.
+    """
+
+    @abstractmethod
+    def calculate(self):
+        """
+        Calculate accuracy metric for model.
+        """
+
+    @property
+    @abstractmethod
+    def sample_generator(self) -> `SampleGenerator`:
+      """
+      :return: The instance of `SampleGenerator`.
+      """
+
```

Comment:
What about classmethod, statismethod, "internal" methods, property setter?

```suggestion
from abc import ABC, abstractmethod

class C(ABC):
    @abstractmethod
    def my_abstract_method(self, ...):
        ...
    @classmethod
    @abstractmethod
    def my_abstract_classmethod(cls, ...):
        ...
    @staticmethod
    @abstractmethod
    def my_abstract_staticmethod(...):
        ...
    @property
    @abstractmethod
    def my_abstract_property(self):
        ...
    @my_abstract_property.setter
    @abstractmethod
    def my_abstract_property(self, val):
        ...
    @abstractmethod
    def _get_x(self):
        ...
    @abstractmethod
    def _set_x(self, val):
```

Reply:
It looks good for me.